### PR TITLE
replace argparse with clap

### PR DESCRIPTION
**Pull request by pogobanane**
Clap: a modern argpaser which does the thinking for you. 

The CLI stays the same except: 
- you can now specify ordered lists of container types `ctnr --type docker,rkt`
- passing arguments to a command requires `--` if command arguments contain `-`-flags. Example: `cntr attach --type docker -- ls -l /dev`

Example:
```
> cargo run -- attach -h
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/cntr attach -h`
cntr-attach
Enter container

USAGE:
    cntr attach [OPTIONS] <id> [command]...

FLAGS:
    -h, --help    Prints help information

OPTIONS:
        --effective-user <EFFECTIVE_USER>    effective username that should be owner of new created files on the host
    -t, --type <TYPE>                        Container types to try (sperated by ','). [default: all but command]
                                             [possible values: process_id, rkt, podman, docker, nspawn, lxc, lxd,
                                             containerd, command]

ARGS:
    <id>            container id, container name or process id
    <command>...    Command and its arguments to execute after attach. Consider prepending it with '-- ' to prevent
                    parsing of '-x'-like flags. [default: $SHELL]
```

### CODE REVIEW

File: src/bin/main.rs [hunk 1]
```diff
@@ -157,39 +89,54 @@ fn exec_command(args: Vec<String>, setcap: bool) {
...
-        Ok(exe) => {
-            if exe == Path::new(cntr::SETCAP_EXE) {
-                exec_command(env::args().collect::<Vec<String>>(), true)
-            }
-        }
-        Err(e) => {
-            eprintln!("failed to resolve executable: {}", e);
-            process::exit(1);
-        }
-    }
```
Comment (Mic92) [inline src/bin/main.rs:170]: This part need to be kept. If the current executable is an capability executable wrapper it should execute the binary right away.

Comment (pogobanane) [reply src/bin/main.rs]: fixed in d05b1be. Should i resolve the merge conflict in Cargo.toml now to allow merge?

**Comment by Mic92** [comment]
I resolved the merge conflict but I think you also need to update `Cargo.lock`.

**Comment by Mic92** [comment]
Mhm my merge was also faulty... `git rebase origin/master` and a force push might be a good idea.

File: src/bin/main.rs [hunk 2]
```diff
@@ -157,39 +89,71 @@ fn exec_command(args: Vec<String>, setcap: bool) {
...
     match std::env::current_exe() {
         Ok(exe) => {
             if exe == Path::new(cntr::SETCAP_EXE) {
-                exec_command(env::args().collect::<Vec<String>>(), true)
+                let matches = exec_command.get_matches();
+                exec(&matches, true);
+            } else {
+                let matches = main_app.get_matches();
+                match matches.subcommand() {
+                    ("exec", Some(exec_matches)) => exec(exec_matches, true),
```
Comment (Mic92) [inline src/bin/main.rs:147]: Almost correct:

```suggestion
                    ("exec", Some(exec_matches)) => exec(exec_matches, false),
```

fixed and merged in https://github.com/Mic92/cntr/pull/44

**Comment by Mic92** [comment]
Thanks merged in https://github.com/Mic92/cntr/pull/44

**Pull request closed without merging**